src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -575,11 +575,26 @@ private void processPackage(DetailAST ast, DetailAST nextToken) {
                 log(ast, MSG_SHOULD_BE_SEPARATED, ast.getText());
             }
         }
+        if (isLineEmptyAfterPackage(ast)) {
+            final DetailAST elementAST = ast.getNextSibling();
+            log(elementAST, MSG_SHOULD_BE_SEPARATED, elementAST.getText());
+        }
         if (!hasEmptyLineAfter(ast)) {
             log(nextToken, MSG_SHOULD_BE_SEPARATED, nextToken.getText());
         }
     }
 
+    /**
+     * Checks if there is another element at next line of package declaration.
+     *
+     * @param ast Package ast.
+     * @return true, if there is a element.
+     */
+    private static boolean isLineEmptyAfterPackage(DetailAST ast) {
```

Comment:
Why is this method called `isLineEmptyAfterPackage`?
It can be used with any other element. This method checks if an element is followed by a blank line. It should be something like `hasEmptyLineAfter`.
Wait, we already have this method! But it is hardcoded to ignore any comments.
This is done for a reason, to allow a Javadoc __before__ package definition.
I really ask you to spend a little time to find out if it is possible to use the `isPrecededByJavadoc` method instead of `isComment` in this method.

Reply:
> Why is this method called isLineEmptyAfterPackage? It can be used with any other element.


No Sir, this can't be used as here we are checking empty line after the `lastChild` of `AST`. But this cannot be used for others as not every AST declaration is as such. Eg - `METHOD_DEF` . 
______________________________________________________________________________________

The already present `hasEmptyLineAfter` is hard coded to ignore comments to allow trailing comments that are valid.
Eg: https://github.com/checkstyle/checkstyle/blob/13481f2c410e4944ecf5ab93ec49948a523a0c82/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparator.java#L20
 I am not able to make any relation with Javadoc comment with this method, please correct me if I am wrong.

Also, I have found some corner cases that I have missed before. I am generating a new report for them. Please check.